EIPS/eip-3155.md

```diff
@@ -0,0 +1,241 @@
+{"output":"","gasUsed":"0x3","time":141485}
+``` 
+
+```json
+{
+ "TraceTest": {
+  "env": {
+   "currentCoinbase": "b94f5374fce5edbc8e2a8697c15331677e6ebf0b",
+   "currentDifficulty": "0x20000",
+   "currentGasLimit": "0x26e1f476fe1e22",
+   "currentNumber": "0x1",
+   "currentTimestamp": "0x3e8",
+   "previousHash": "0x0000000000000000000000000000000000000000000000000000000000000000"
+  },
+  "pre": {
+   "0x00000000000000000000000000000ca1100b1a7e": {
+    "code": "0x6040",
+    "storage": {},
+    "balance": "0x0",
+    "nonce": "0x0"
+   },
+   "0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b": {
+    "code": "0x",
+    "storage": {},
+    "balance": "0xffffffff",
+    "nonce": "0x0"
+   }
+  },
+  "transaction": {
+   "gasPrice": "0x1",
+   "nonce": "0x0",
+   "to": "0x00000000000000000000000000000Ca1100b1A7E",
+   "data": [
+    "0x80e3193e421154d1de1cd3a0b425cc21eaa184eb2ec89a756a8e4624"
+   ],
+   "gasLimit": [
+    "0x7a1200"
+   ],
+   "value": [
+    "0x4862"
+   ],
+   "secretKey": "0x45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8"
+  },
+  "out": "0x",
+  "post": {
+   "Istanbul": [
+    {
+     "hash": "3f7878bb9cd21378f7eb0a2d26e24f11abca4709482cc92e31c1a8a9e4aeeed5",
+     "logs": "1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
+     "indexes": {
+      "data": 0,
+      "gas": 0,
+      "value": 0
+     }
+    }
+   ], 
+   "Berlin": [
+    {
+     "hash": "3f7878bb9cd21378f7eb0a2d26e24f11abca4709482cc92e31c1a8a9e4aeeed5",
+     "logs": "1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
+     "indexes": {
+      "data": 0,
+      "gas": 0,
+      "value": 0
+     }
+    }
+   ]
+  }
+ }
+}
+```
+
+```bash
+$evm --nomemory --json statetest test.json
+{"pc":0,"op":96,"gas":"0x79be38","gasCost":"0x3","memory":"0x","memSize":0,"stack":[],"returnStack":[],"returnData":"0x","depth":1,"refund":0,"opName":"PUSH1","error":""}
+{"pc":2,"op":0,"gas":"0x79be35","gasCost":"0x0","memory":"0x","memSize":0,"stack":["0x40"],"returnStack":[],"returnData":"0x","depth":1,"refund":0,"opName":"STOP","error":""}
+{"output":"","gasUsed":"0x3","time":166351}
+{"stateRoot": "3f7878bb9cd21378f7eb0a2d26e24f11abca4709482cc92e31c1a8a9e4aeeed5"}
+{"pc":0,"op":96,"gas":"0x79be38","gasCost":"0x3","memory":"0x","memSize":0,"stack":[],"returnStack":[],"returnData":"0x","depth":1,"refund":0,"opName":"PUSH1","error":""}
+{"pc":2,"op":0,"gas":"0x79be35","gasCost":"0x0","memory":"0x","memSize":0,"stack":["0x40"],"returnStack":[],"returnData":"0x","depth":1,"refund":0,"opName":"STOP","error":""}
+{"output":"","gasUsed":"0x3","time":50655}
+{"stateRoot": "3f7878bb9cd21378f7eb0a2d26e24f11abca4709482cc92e31c1a8a9e4aeeed5"}
+[
+  {
+    "name": "TraceTest",
+    "pass": true,
+    "fork": "Istanbul"
+  },
+  {
+    "name": "TraceTest",
+    "pass": true,
+    "fork": "Berlin"
+  }
+]
+```
+
+</details>
+
+
+## Implementation
```

Comment:
Note, this section recently became optional. I recommend removing it as we prefer to not have external links in EIPs.

Reply:
I'm not sure about this. I think it would be good for people to have a look at implementation examples if they want to implement it in their client

Reply:
Implementation examples are good, just not appropriate for a technical specification document.  EIPs should not be seen as a tutorial or educational material, such content is much better suited for blogs, programming websites, etc.